src/webui/api/torrentscontroller.cpp

```diff
@@ -113,17 +118,91 @@ namespace
     void applyToTorrents(const QStringList &hashes, const std::function<void (BitTorrent::TorrentHandle *torrent)> &func)
     {
         if ((hashes.size() == 1) && (hashes[0] == QLatin1String("all"))) {
-            foreach (BitTorrent::TorrentHandle *torrent, BitTorrent::Session::instance()->torrents())
+            foreach (BitTorrent::TorrentHandle *const torrent, BitTorrent::Session::instance()->torrents())
                 func(torrent);
         }
         else {
             for (const QString &hash : hashes) {
-                BitTorrent::TorrentHandle *torrent = BitTorrent::Session::instance()->findTorrent(hash);
+                BitTorrent::TorrentHandle *const torrent = BitTorrent::Session::instance()->findTorrent(hash);
                 if (torrent)
                     func(torrent);
             }
         }
     }
+
+    QVariantList getStickyTrackers(const BitTorrent::TorrentHandle *const torrent)
```

Comment:
It seems we can make it a few lines shorter without sacrificing too much readability:
```c++
    QVariantList getStickyTrackers(const BitTorrent::TorrentHandle *const torrent)
    {
        uint seedsDHT = 0, seedsPeX = 0, seedsLSD = 0, leechesDHT = 0, leechesPeX = 0, leechesLSD = 0;
        for (const BitTorrent::PeerInfo &peer : torrent->peers()) {
            if (peer.isConnecting()) continue;

            if (peer.fromDHT()) {
                if (peer.isSeed())
                    ++seedsDHT;
                else
                    ++leechesDHT;
            }
            if (peer.fromPeX()) {
                if (peer.isSeed())
                    ++seedsPeX;
                else
                    ++leechesPeX;
            }
            if (peer.fromLSD()) {
                if (peer.isSeed())
                    ++seedsLSD;
                else
                    ++leechesLSD;
            }
        }

        const QString working {QCoreApplication::translate("TrackerListWidget", "Working")};
        const QString disabled {QCoreApplication::translate("TrackerListWidget", "Disabled")};
        const QString privateMsg {QCoreApplication::translate("TrackerListWidget", "This torrent is private")};

        const QVariantMap dht {
            {KEY_TRACKER_URL, "** [DHT] **"},
            {KEY_TRACKER_TIER, ""},
            {KEY_TRACKER_MSG, (torrent->isPrivate() ? privateMsg : "")},
            {KEY_TRACKER_STATUS, ((BitTorrent::Session::instance()->isDHTEnabled() && !torrent->isPrivate()) ? working : disabled)},
            {KEY_TRACKER_PEERS_COUNT, 0},
            {KEY_TRACKER_DOWNLOADED_COUNT, 0},
            {KEY_TRACKER_SEEDS_COUNT, seedsDHT},
            {KEY_TRACKER_LEECHES_COUNT, leechesDHT}
        };

        const QVariantMap pex {
            {KEY_TRACKER_URL, "** [PeX] **"},
            {KEY_TRACKER_TIER, ""},
            {KEY_TRACKER_MSG, (torrent->isPrivate() ? privateMsg : "")},
            {KEY_TRACKER_STATUS, ((BitTorrent::Session::instance()->isPeXEnabled() && !torrent->isPrivate()) ? working : disabled)},
            {KEY_TRACKER_PEERS_COUNT, 0},
            {KEY_TRACKER_DOWNLOADED_COUNT, 0},
            {KEY_TRACKER_SEEDS_COUNT, seedsPeX},
            {KEY_TRACKER_LEECHES_COUNT, leechesPeX}
        };

        const QVariantMap lsd {
            {KEY_TRACKER_URL, "** [LSD] **"},
            {KEY_TRACKER_TIER, ""},
            {KEY_TRACKER_MSG, (torrent->isPrivate() ? privateMsg : "")},
            {KEY_TRACKER_STATUS, ((BitTorrent::Session::instance()->isLSDEnabled() && !torrent->isPrivate()) ? working : disabled)},
            {KEY_TRACKER_PEERS_COUNT, 0},
            {KEY_TRACKER_DOWNLOADED_COUNT, 0},
            {KEY_TRACKER_SEEDS_COUNT, seedsLSD},
            {KEY_TRACKER_LEECHES_COUNT, leechesLSD}
        };

        return QVariantList {dht, pex, lsd};
    }
```

Reply:
This version looks much cleaner. Will change